packages/terra-date-picker/src/DateInput.jsx

```diff
@@ -199,7 +209,7 @@ class DatePickerInput extends React.Component {
           placeholder={placeholder}
           onFocus={onFocus}
           onBlur={onBlur}
-          aria-label={intl.formatMessage({ id: 'Terra.datePicker.date' })}
+          ariaLabel={value ? `${label}, ${getLocalizedDateForScreenReader(DateUtil.createSafeDate(dateValue), this.props)}` : label}
```

Comment:
`getLocalizedDateForScreenReader` expects that the props object contains both [`intl` and `locale`](https://github.com/cerner/terra-framework/blob/4880772b8b92bb05f99ef3f3084a8d1e4c389bdc/packages/terra-date-picker/src/react-datepicker/date_utils.js#L381). `this.props` here doesn't have `locale`. You may need to get the locale from `this.props.intl.locale` and create a new props object like [here](https://github.com/cerner/terra-framework/blob/4880772b8b92bb05f99ef3f3084a8d1e4c389bdc/packages/terra-date-picker/src/react-datepicker/month.jsx#L237).

Reply:
Updated here https://github.com/cerner/terra-framework/pull/1182/commits/acc7fb02dee7e9f4e811573c8c32587215dd2517.